packages/blocknumber/src/providers/blockmetaJsonBlockNumberProvider.ts

```diff
@@ -0,0 +1,126 @@
+import { ILogger, Timestamp } from "@ebo-agent/shared";
+import axios, { AxiosInstance, AxiosResponse, isAxiosError } from "axios";
+
+import { UndefinedBlockNumber } from "../exceptions/undefinedBlockNumber.js";
+import { BlockNumberProvider } from "./blockNumberProvider.js";
+
+type BlockByTimeResponse = {
+    num: string;
+    id: string;
+    time: string;
+};
+
+export type BlockmetaClientConfig = {
+    baseUrl: URL;
+    servicePath: string;
+    bearerToken: string;
+};
+
+/**
+ * Consumes the blockmeta.BlockByTime substreams' service via HTTP POST JSON requests to provide
+ * block numbers based on timestamps
+ *
+ * Refer to these web pages for more information:
+ * * https://thegraph.market/
+ * * https://substreams.streamingfast.io/documentation/consume/authentication
+ */
+export class BlockmetaJsonBlockNumberProvider implements BlockNumberProvider {
+    private readonly axios: AxiosInstance;
+
+    constructor(
+        private readonly options: BlockmetaClientConfig,
+        private readonly logger: ILogger,
+    ) {
+        const { baseUrl, bearerToken } = options;
+
+        this.axios = axios.create({
+            baseURL: baseUrl.toString(),
+            headers: {
+                common: {
+                    "Content-Type": "application/json",
+                    Authorization: `Bearer ${bearerToken}`,
+                },
+            },
+        });
+    }
+
+    /** @inheritdoc */
+    async getEpochBlockNumber(timestamp: Timestamp): Promise<bigint> {
+        if (timestamp > Number.MAX_SAFE_INTEGER || timestamp < Number.MIN_SAFE_INTEGER)
```

Comment:
```suggestion
        if (timestamp > Number.MAX_SAFE_INTEGER || timestamp < 0)
```

Number.MIN_SAFE_INTEGER = -9007199254740991

Reply:
Just for fun I've requested the `BlockByTime/After` after `'{"time": "1950-01-01T00:00:00.000Z"}'` and got:
```
"num":"20737620", "time":"1915-04-21T01:10:00.999Z"
```


There's probably a weird overflow bug in substreams lol should we report it? Doesn't seem too big of an issue as no sane person would try to fetch blocks before computers even existed

Reply:
Hmmm, maybe we should just tell the graph team on slack